Remotion/Data/DomainObjects/Persistence/Rdbms/SqlServer/Parameters/SqlTableValuedParameterValue.cs

```diff
@@ -82,5 +82,84 @@ IEnumerator<SqlDataRecord> IEnumerable<SqlDataRecord>.GetEnumerator ()
   }
 
   public int Count => _records.Count;
+
+  public bool Equals (SqlTableValuedParameterValue? other)
+  {
+    if (other == null)
+      return false;
+
+    if (TableTypeName != other.TableTypeName)
+      return false;
+
+    if (_columnMetaData.Length != other._columnMetaData.Length)
+      return false;
+
+    if (_records.Count != other._records.Count)
+      return false;
+
+    for (var i = 0; i < _columnMetaData.Length; i++)
+    {
+      var ours = _columnMetaData[i];
+      var theirs = other._columnMetaData[i];
+      if (!theirs.Name.Equals(ours.Name))
+        return false;
+
+      if (!theirs.DbType.Equals(ours.DbType))
+        return false;
+
+      if (!theirs.MaxLength.Equals(ours.MaxLength))
+        return false;
```

Comment:
please use `==` instead of `.Equals`
reasons: 
earlier, you already use `!=`
DbType.Equals uses the Object overload and thus boxing.